### PR TITLE
fix: enforce Azure CNI config via jq instead of sed

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR reduces the bash complexity of the jq usage inside the Azure CNI setup foo so that the Azure CNI configuration is always a multi-line file to accommodate all the pre-existing (known-working) sed foo.

A future workstream would be to replace the sed foo with jq foo.

Before:

```
azureuser@k8s-agentpool1-40173905-vmss000001:~$ echo $(sudo cat "/etc/cni/net.d/10-azure.conflist" | jq '.plugins[0].ipv6Mode="ipv6nat"') > test.json
azureuser@k8s-agentpool1-40173905-vmss000001:~$ cat test.json 
{ "cniVersion": "0.3.0", "name": "azure", "plugins": [ { "type": "azure-vnet", "mode": "transparent", "bridge": "azure0", "ipsToRouteViaHost": [ "169.254.20.10" ], "ipam": { "type": "azure-vnet-ipam" }, "ipv6Mode": "ipv6nat" }, { "type": "portmap", "capabilities": { "portMappings": true }, "snat": true } ] }
```

In other words, the resultant JSON is a string with no newlines (which the sed commands aren't able to interpret).

After:

```
azureuser@k8s-agentpool1-40173905-vmss000001:~$ sudo jq '.plugins[0].ipv6Mode="ipv6nat"' "/etc/cni/net.d/10-azure.conflist" > test.json
azureuser@k8s-agentpool1-40173905-vmss000001:~$ cat test.json 
{
  "cniVersion": "0.3.0",
  "name": "azure",
  "plugins": [
    {
      "type": "azure-vnet",
      "mode": "transparent",
      "bridge": "azure0",
      "ipsToRouteViaHost": [
        "169.254.20.10"
      ],
      "ipam": {
        "type": "azure-vnet-ipam"
      },
      "ipv6Mode": "ipv6nat"
    },
    {
      "type": "portmap",
      "capabilities": {
        "portMappings": true
      },
      "snat": true
    }
  ]
}
```

This PR also rationalizes the ipv6 dual stack implementation and configuration, which requires bridge mode in order to work.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -271,15 +271,18 @@ configureCNI() {
...
 }
 configureAzureCNI() {
+  local tmpDir=$(mktemp -d "$(pwd)/XXX")
   if [[ "${NETWORK_PLUGIN}" == "azure" ]]; then
     mv $CNI_BIN_DIR/10-azure.conflist $CNI_CONFIG_DIR/
     chmod 600 $CNI_CONFIG_DIR/10-azure.conflist
-    if [[ "${IPV6_DUALSTACK_ENABLED}" == "true" ]]; then
-      echo $(cat "$CNI_CONFIG_DIR/10-azure.conflist" | jq '.plugins[0].ipv6Mode="ipv6nat"') > "$CNI_CONFIG_DIR/10-azure.conflist"
-    fi
+{{- if IsIPv6DualStackFeatureEnabled}}
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_config.sh:278]: saving cloud-init payload here

File: parts/k8s/cloud-init/artifacts/cse_config.sh [hunk 2]
```diff
@@ -271,21 +271,26 @@ configureCNI() {
...
+{{- end}}
     if [[ {{GetKubeProxyMode}} == "ipvs" ]]; then
       serviceCidrs={{GetServiceCidr}}
-      echo $(cat "$CNI_CONFIG_DIR/10-azure.conflist" | jq  --arg serviceCidrs $serviceCidrs '.plugins[0]+={serviceCidrs: $serviceCidrs}') > /etc/cni/net.d/10-azure.conflist
+      jq --arg serviceCidrs $serviceCidrs '.plugins[0]+={serviceCidrs: $serviceCidrs}' "$CNI_CONFIG_DIR/10-azure.conflist" > $tmpDir/tmp
+      mv $tmpDir/tmp $CNI_CONFIG_DIR/10-azure.conflist
     fi
     if [[ "${NETWORK_MODE}" == "bridge" ]]; then
-      sed -i 's#"mode":"transparent"#"mode":"bridge"#g' $CNI_CONFIG_DIR/10-azure.conflist
+      jq '.plugins[0].mode="bridge"' "$CNI_CONFIG_DIR/10-azure.conflist" > $tmpDir/tmp
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_config.sh:288]: Every time you replace a sed with a jq a @jchauncey gets his wings

File: test/e2e/test_cluster_configs/everything.json [hunk 3]
```diff
@@ -14,6 +14,8 @@
 		"properties": {
 			"orchestratorProfile": {
 				"kubernetesConfig": {
+					"networkPlugin": "azure",
```
Comment (jackfrancis) [inline test/e2e/test_cluster_configs/everything.json:17]: These changes ensure we still have test coverage for Azure CNI + bridge mode now that Azure CNI 1.2.0 has merged with transparent mode enabled by default.

**Review by aramase — COMMENTED** [review]
/lgtm

**Pull request merged**